test/unit/blocks_control.js

```diff
@@ -0,0 +1,130 @@
+var test = require('tap').test;
+var Control = require('../../src/blocks/scratch3_control');
+var Runtime = require('../../src/engine/runtime');
+
+test('getPrimitives', function (t) {
+    var rt = new Runtime();
+    var c = new Control(rt);
+    t.type(c.getPrimitives(), 'object');
+    t.end();
+});
+
+test('repeat', function (t) {
+    var rt = new Runtime();
+    var c = new Control(rt);
+
+    // Test harness (mocks `util`)
+    var i = 0;
+    var repeat = 10;
+    var util = {
+        stackFrame: Object.create(null),
+        startBranch: function () {
+            i++;
+            c.repeat({TIMES: repeat}, util);
```

Comment:
This makes sense that it works but feels weird that `startBranch` is going to call `c.repeat` again. In reality `startBranch` would set up the sequencer so that we get something more like `for (var i = 0; i < 10; i++) { c.repeat{TIMES: 10}, util);}` if that makes any sense. However I'm fine with this as a unit test, I suppose...